pkg/kubelet/token/token_manager.go

```diff
@@ -146,6 +149,14 @@ func (m *Manager) requiresRefresh(tr *authenticationv1.TokenRequest) bool {
 }
 
 // keys should be nonconfidential and safe to log
-func keyFunc(name, namespace string, tr *authenticationv1.TokenRequest) string {
-	return fmt.Sprintf("%q/%q/%#v", name, namespace, tr.Spec)
+func keyFunc(name, namespace string, tr *authenticationv1.TokenRequest) (string, error) {
+	if tr.Spec.ExpirationSeconds == nil {
```

Comment:
I don't think this should return error just because dereferencing the pointers would panic. Declare vars with empty values for these fields and set them if fields aren't nil.